Cargo.toml

```diff
@@ -214,7 +214,7 @@ tokio-postgres = { version = "0.5.5", optional = true, features = ["runtime", "w
 # Indirect dependency; pinning until
 # https://github.com/timberio/vector/issues/6005 is resolved
 thread_local = "= 1.0.1"
-dashmap = "3"
+dashmap = "4"
```

Comment:
ISTR there was a regression issue with dashmap 4, though I don't recall the details. Perhaps relevant to the metrics performance regression?

Reply:
It is exactly my thinking as well - `dashmap` 4 has a different and slower implementation, and I suspect this to be the cause of the regression.

Reply:
Downgrade `dashmap` back to 3.

Reply:
@fanatid do you think this was the cause of the regression you [noted here](https://github.com/timberio/vector/pull/6217#issuecomment-766435360)? If so, we might be able to upgrade the `metrics` dependency?

Reply:
No, it does not help. I downgraded metrics crates in addition and waiting benchmark for confirmation that everything is ok.